test/rulesets/Base/indexing.jl

```diff
@@ -48,3 +48,23 @@
         end
     end
 end
+
+if VERSION >= v"1.6"
+    # Without these, you get errors, e.g.
+    # test_rrule: eachrow on Vector{Float64}: Error During Test at /Users/me/.julia/packages/ChainRulesTestUtils/8dFTY/src/testers.jl:195
+    #   Got exception outside of a @test
+    #   DimensionMismatch("second dimension of A, 6, does not match length of x, 5")
+    ChainRules.rrule(::typeof(collect∘eachrow), x) = rrule(eachrow, x)
+    ChainRules.rrule(::typeof(collect∘eachcol), x) = rrule(eachcol, x)
+    ChainRules.rrule(::typeof(collect∘eachslice), x; dims) = rrule(eachslice, x; dims=dims)
+    # And before Julia 1.6, `collect∘eachrow` is not a specific type.
```

Comment:
I wonder if we should push these into the `src` also behind this conditional
with a comment saying that we always collect eachrow etc anyway.

Reply:
Yes that seems fine, too.

Reply:
Done.